etc/cmake/options.cmake

```diff
@@ -321,6 +321,7 @@ ot_multi_option(OT_LOG_OUTPUT OT_LOG_OUTPUT_VALUES OPENTHREAD_CONFIG_LOG_OUTPUT
 
 # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 
+ot_string_option(OT_NETDIAG_VENDOR_INFO OPENTHREAD_CONFIG_NET_DIAG_VENDOR_INFO_SET_API_ENABLE "Allow setting vendor info at runtime")
```

Comment:
This should not be a "string option". It is a normal ON/OFF option, so should use `ot_option()` for it.
Also suggest moving this within the list of `ot_option()` definitions above in this file (the list is sorted).